backend/business-partner-agent/src/main/java/org/hyperledger/bpa/config/ActivityLogConfig.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hyperledger.bpa.config;
+
+import lombok.NoArgsConstructor;
+import org.hyperledger.aries.api.connection.ConnectionState;
+import org.hyperledger.aries.api.present_proof.PresentationExchangeState;
+
+import javax.inject.Singleton;
+import java.util.List;
+
+@Singleton
+@NoArgsConstructor
+public class ActivityLogConfig {
+    /*
+     * For now, we are not sure what to do with this configuration. Will each
+     * installation determine what their own definition of "complete" is, or which
+     * events will be "tasks"? And tasks are going to be dependant in the
+     * AcaPyConfig, which flags are set to auto respond...
+     */
+
+    public List<ConnectionState> getConnectionStatesForActivities() {
+        return connectionStates(ConnectionState.REQUEST, ConnectionState.INVITATION, ConnectionState.ACTIVE,
+                ConnectionState.RESPONSE);
+    }
+
+    public List<ConnectionState> getConnectionStatesForTasks() {
```

Comment:
```java
if (acaPyConfig.getAutoAcceptRequests() != null && acaPyConfig.getAutoAcceptRequests() ) {
    return List.of();
}
return connectionStates(ConnectionState.REQUEST);

```

Like this there is no event and it is also filtered in the manager.

Reply:
Very nice. Very clean.